pkg/agent/monitortool/monitor.go

```diff
@@ -0,0 +1,469 @@
+	// isIPv6Enabled is the flag to indicate whether the IPv6 is enabled.
+	isIPv6Enabled bool
+
+	// nodeName is the name of the current Node, used to filter out the current Node from the latency monitor.
+	nodeName                   string
+	nodeInformer               coreinformers.NodeInformer
+	nodeLatencyMonitorInformer crdinformers.NodeLatencyMonitorInformer
+}
+
+// LatencyConfig is the config for the latency monitor.
+type LatencyConfig struct {
+	// Enable is the flag to enable the latency monitor.
+	Enable bool
+	// Interval is the interval time to ping all Nodes.
+	Interval time.Duration
+}
+
+// NewNodeLatencyMonitor creates a new NodeLatencyMonitor.
+func NewNodeLatencyMonitor(nodeInformer coreinformers.NodeInformer,
+	nlmInformer crdinformers.NodeLatencyMonitorInformer,
+	nodeConfig *config.NodeConfig,
+	trafficEncapMode config.TrafficEncapModeType) *NodeLatencyMonitor {
+	m := &NodeLatencyMonitor{
+		latencyStore:               NewLatencyStore(trafficEncapMode.IsNetworkPolicyOnly()),
+		latencyConfigChanged:       make(chan LatencyConfig),
+		nodeInformer:               nodeInformer,
+		nodeLatencyMonitorInformer: nlmInformer,
+		nodeName:                   nodeConfig.Name,
+	}
+
+	m.isIPv4Enabled, _ = config.IsIPv4Enabled(nodeConfig, trafficEncapMode)
+	m.isIPv6Enabled, _ = config.IsIPv6Enabled(nodeConfig, trafficEncapMode)
+
+	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    m.onNodeAdd,
+		UpdateFunc: m.onNodeUpdate,
+		DeleteFunc: m.onNodeDelete,
+	})
+
+	nlmInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    m.onNodeLatencyMonitorAdd,
+		UpdateFunc: m.onNodeLatencyMonitorUpdate,
+		DeleteFunc: m.onNodeLatencyMonitorDelete,
+	})
+
+	return m
+}
+
+// Is current node
+func (m *NodeLatencyMonitor) isCurrentNode(node *corev1.Node) bool {
+	return node.Name == m.nodeName
+}
+
+// onNodeAdd is the event handler for adding Node.
+func (m *NodeLatencyMonitor) onNodeAdd(obj interface{}) {
+	node := obj.(*corev1.Node)
+	if m.isCurrentNode(node) {
+		return
+	}
+
+	m.latencyStore.addNode(node)
+
+	klog.V(4).InfoS("Node added", "Node", klog.KObj(node))
+}
+
+// onNodeUpdate is the event handler for updating Node.
+func (m *NodeLatencyMonitor) onNodeUpdate(oldObj, newObj interface{}) {
+	node := newObj.(*corev1.Node)
+	if m.isCurrentNode(node) {
+		return
+	}
+
+	m.latencyStore.updateNode(node)
+
+	klog.V(4).InfoS("Node updated", "Node", klog.KObj(node))
+}
+
+// onNodeDelete is the event handler for deleting Node.
+func (m *NodeLatencyMonitor) onNodeDelete(obj interface{}) {
+	node, ok := obj.(*corev1.Node)
+	if !ok {
+		deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.ErrorS(nil, "Received unexpected object", "obj", obj)
+			return
+		}
+		node, ok = deletedState.Obj.(*corev1.Node)
+		if !ok {
+			klog.ErrorS(nil, "DeletedFinalStateUnknown contains non-Node object", "obj", deletedState.Obj)
+			return
+		}
+
+		if m.isCurrentNode(node) {
```

Comment:
it should be moved to out of the if !ok branch.

Reply:
OK! I have fixed it.